lib/galaxy/datatypes/media.py

```diff
@@ -198,30 +287,14 @@ class Mp4(Video):
     file_ext = "mp4"
 
     def sniff(self, filename: str) -> bool:
-        if which("ffprobe"):
-            metadata, streams = ffprobe(filename)
-            return "mp4" in metadata["format_name"].split(",")
-        return False
+        return _get_file_format_from_magic_number(filename, "mp4")
```

Comment:
is there a reason to *remove* fallback here? (and other places) I think this would be better if you just replaced the `return False` line, if we've got ffprobe, let's use it and feel certain we're detecting 100% correctly even in 'odd' cases.

Reply:
So, should we always look for `ffprobe` and if it doesn't work check with the magic number?

Reply:
> So, should we always look for ffprobe and if it doesn't work check with the magic number?

Yes, not everyone is going to install ffmpeg, and your magic numbesr are a good, efficient fallback. (Unless any other devs want to chime in here?)

> Also when using 'ffprobe' on 'mov' and 'mp4' files, it gives back the same result. One way to tell them apart is by looking at the magic number.

Thank you for pointing this out, you are correct. When I was looking further into it, I notice this output, ffmpeg bundles a lot of formats together on that line.

```
{
    "format": {
        "filename": "sample-5s.mp4",
        "nb_streams": 2,
        "nb_programs": 0,
        "format_name": "mov,mp4,m4a,3gp,3g2,mj2",
        "format_long_name": "QuickTime / MOV",
```

which is an interesting result. Probably due to this: https://en.wikipedia.org/wiki/QuickTime_File_Format#Relation_to_MP4

Which explains why ffmpeg will decode both of those with the same decoder (since it's an mp4 stream internally with slightly different header metadata), but will encode them as separate spec compliant containers.

```
  E mov             QuickTime / MOV
 D  mov,mp4,m4a,3gp,3g2,mj2 QuickTime / MOV
```

I wonder how much of a distinction we need to make here, for some relatively ancient formats, but probably some microscope/image tooling cares I guess.

As for implementation, I'm not sure what's the optimal implementation here. magic numbers can confirm it's actually one of the datatypes, but I still have some (perhaps unfounded) concern that they are not fully exhaustive/forward compatible of all the flavours of these datatypes that can be seen in the wild, so, instead if ffprobe says it's mp4, then we'll accept that.

Given how that magic numbers are used to spell `ftypisom` / `ftypMSNV`, I think that worry is not unfounded, that they'll might eventually a new `ftyp` that's also somehow an mp4 file.

So I think the optimal implementation looks like:

in mp4:
- if ffprobe says it's an mp4
    - if magic numbers say it's a mov, return false
    - otherwise it's probably an mp4
in mov
- if the magic numbers say it's a mov it's a mov.

and based on their ordering in the datatypes to prefer mp4 to mov/qtff (given that it's a more common format).

Reply:
Why is the ffprobe better than the magic bits? The idea is to cut down on the number of subprocess calls on the headnodes, which is a good thing. So, should we use just magic bits if it is possible to eliminate ffprobe subprocess to fasten the process?

Reply:
> The idea is to cut down on the number of subprocess calls on the headnodes, which is a good thing

We currently only do 1 subprocess call (or we should only be doing 1) for any media file due to the caching in this function: https://github.com/galaxyproject/galaxy/blob/d7673ab04bb5b627297e849580e149584d633d7a/lib/galaxy/datatypes/media.py#L22-L33

so this will not achieve any improvement there. Did you have a metric which was showing a high number of subprocess calls? Perhaps the cache size needs to be larger (or configurable)